www/js/draw-model.js

```diff
@@ -530,14 +539,12 @@ define([
 				domConstruct.destroy(ID);
 			}
 			domConstruct.destroy(nodeID);
-			if(this._model.getColor(ID) !== undefined){
-				if(type === 'quantity'){
-					this._quantityNodeCount--;
-				}else if( type === 'equation'){
-					this._equationNodeCount--;
-				}
-				this.addNodeCount();
+			if(type === 'quantity' && this._model.getVariableType(ID) === "unknown"){
```

Comment:
you should be checking for dynamic as well in this. That field will be coming back in the code.